src/Common/src/System/IO/PathInternal.Windows.cs

```diff
@@ -70,11 +71,27 @@ internal static bool IsDirectoryTooLong(string fullPath)
         }
 
         /// <summary>
-        /// Adds the extended path prefix (\\?\) if not already present.
+        /// Adds the extended path prefix (\\?\) if not already present, IF the path is not relative,
+        /// AND the path is more than 259 characters. (> MAX_PATH + null)
+        /// </summary>
+        internal static string AddExtendedPathPrefixForLongPaths(string path)
+        {
+            if (path != null && path.Length >= MaxShortPath)
+            {
+                return AddExtendedPathPrefix(path);
+            }
+            else
+            {
+                return path;
+            }
+        }
+
+        /// <summary>
+        /// Adds the extended path prefix (\\?\) if not already present, IF the path is not relative.
         /// </summary>
         internal static string AddExtendedPathPrefix(string path)
         {
-            if (IsExtended(path))
+            if (IsExtended(path) || IsPathRelative(path) || IsDevicePath(path))
```

Comment:
Does anyone call this function beyond the above one that does it conditionally?  If not, should this be private?  It's also strange to me that we have an "Add" method that implies it adds something unconditionally but then has all of these conditions at the beginning.  Folding it into the "IfNecessary" version would help with that disconnect.